spec/VolumesSpec.php

```diff
@@ -36,34 +37,34 @@ function it_should_provide_search()
     {
         $this->init();
         $res = $this->search('isbn:0253324009');
-        $res->shouldHaveType(\Generator::class);
+        $res->shouldHaveType(LibraryBuilder::class);
     }
 
     function it_should_provide_search_based_lookup()
     {
         $this->init('volumes.json', 'volume.json');
         $res = $this->firstOrNull('isbn:0253324009');
-        $res->shouldHaveType(Volume::class);
+        $res->shouldHaveType(\stdclass::class);
```

Comment:
Hm, why this change?

Reply:
so a the `Volume` class uses `LibraryBuilder` methods which only care about the data that is fetched. I can make it return a volume class but then it might look more like:

```php
$books->volumes->search('hey')->all()->data;
```

the all() just cares about the data property which returns an standard object

Reply:
Yeah, but it makes more sense that the`Volumes` class returns `Volume`, doesn't it? What about injecting the desired class into LibraryBuilder, smtn like `new LibraryBuilder($this->client, Volume::class))`?

Reply:
out of curiosity, what would be the benefit of returning a `Volume` class? are you trying to gain access to additional methods after the api call is fetched?

Reply:
Yeah, convenience methods such as `$volume->getCover()`. We could still make the class traversable so you can use foreach. See #5 (based on master, but shouldn't have many conflicts with your branch)